### PR TITLE
slither

### PR DESCRIPTION
**Pull request by thedavidmeister**

### CODE REVIEW

File: cli/src/meta/mod.rs [hunk 1]
```diff
@@ -355,100 +361,129 @@ impl<'de> Deserialize<'de> for RainMetaDocumentV1Item {
...
     }
-    let request_body = query::MetaQuery::build_query(query::meta_query::Variables {
-        hash: Some(hash.to_ascii_lowercase()),
-    });
+    let query_string = get_meta_query(hash);
     let mut promises = vec![];
     let client = Arc::new(
         Client::builder()
-            .timeout(std::time::Duration::from_secs(timeout as u64))
+            // .timeout(std::time::Duration::from_secs(timeout as u64))
```
Comment (thedavidmeister) [inline cli/src/meta/mod.rs:373]: delete commented out code

File: cli/src/meta/mod.rs [hunk 2]
```diff
@@ -355,100 +361,129 @@ impl<'de> Deserialize<'de> for RainMetaDocumentV1Item {
...
     }
-    let request_body = query::DeployerQuery::build_query(query::deployer_query::Variables {
-        hash: Some(hash.to_ascii_lowercase()),
-    });
+    let query_string = get_deployer_query(hash);
     let mut promises = vec![];
     let client = Arc::new(
         Client::builder()
-            .timeout(std::time::Duration::from_secs(timeout as u64))
+            // .timeout(std::time::Duration::from_secs(timeout as u64))
```
Comment (thedavidmeister) [inline cli/src/meta/mod.rs:399]: delete

File: cli/src/meta/mod.rs [hunk 3]
```diff
@@ -685,37 +799,51 @@ impl Store {
...
     }
 
-    /// updates the meta cache by the given hash and meta bytes, checks the hash to bytes validity
-    /// returns the reference to the authoring bytes if the updated meta bytes contained any
-    pub fn update_with(&mut self, hash: &String, bytes: &Vec<u8>) -> Option<&Vec<u8>> {
-        let mut am_bytes: Option<&Vec<u8>> = None;
+    /// updates the meta cache by the given hash and meta bytes, checks the hash to bytes
+    /// validity returns the reference to the bytes if the updated meta bytes contained any
+    pub fn update_with(&mut self, hash: &str, bytes: &[u8]) -> Option<&Vec<u8>> {
+        // let mut am_bytes: Option<&Vec<u8>> = None;
```
Comment (thedavidmeister) [inline cli/src/meta/mod.rs:830]: delete

File: cli/src/meta/query/mod.rs [hunk 4]
```diff
@@ -1,81 +1,203 @@
...
+        } else {
+            None
+        }
+    }
+}
+
+/// get deployer query string
+pub fn get_deployer_query(hash: &str) -> String {
+    format!(
+        r#"query DeployerQuery {{ expressionDeployers( first: 1
```
Comment (thedavidmeister) [inline cli/src/meta/query/mod.rs:57]: why are the queries in strings?

File: cli/src/meta/mod.rs (hunk 1 above)
Comment (rouzwelt) [reply cli/src/meta/mod.rs:373]: this should be a TODO

Comment (rouzwelt) [reply cli/src/meta/mod.rs:373]: when reqwest adds support for timouts for Clients

File: subgraph/flake.nix [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            mkdir -p contracts && cp -r ../src/* contracts
+
+            npx hardhat compile --force
+
+            rm -rf ./abis && mkdir ./abis
+            cp artifacts/contracts/concrete/MetaBoard.sol/MetaBoard.json abis
+          '');
+
+          ci-test = pkgs.writeShellScriptBin "ci-test" (''
+            npx mustache config/localhost.json subgraph.template.yaml subgraph.yaml
```
Comment (thedavidmeister) [inline subgraph/flake.nix:28]: why are we using npx in nix?

File: cli/src/subgraph/mod.rs [hunk 6]
```diff
@@ -1,45 +1,53 @@
...
 
-    /// Rain NativeParser subgraphs of all implemented networks
+    /// Rain NPE2 subgraphs of all suppoerted networks
+    pub const NPE2: [&'static str; 3] = [Self::ETHEREUM[2], Self::POLYGON[2], Self::MUMBAI[2]];
+
+    /// Rain NativeParser subgraphs of all suppoerted networks
     pub const NP: [&'static str; 3] = [Self::ETHEREUM[1], Self::POLYGON[1], Self::MUMBAI[1]];
 
-    /// Rain legacy(non NativeParser) subgraphs of all implemented networks
+    /// Rain legacy(non NativeParser) subgraphs of all suppoerted networks
```
Comment (thedavidmeister) [inline cli/src/subgraph/mod.rs:33]: typo

File: cli/src/subgraph/mod.rs [hunk 7]
```diff
@@ -1,45 +1,53 @@
-/// # Rain Subgraphs
-/// all known subgraph endpoints
+/// All known subgraph endpoints
 #[derive(Debug, Clone)]
 pub struct KnownSubgraphs;
```
Comment (thedavidmeister) [inline cli/src/subgraph/mod.rs:3]: why is this a struct instead of an enum?

File: cli/src/meta/types/interpreter_caller/v1.rs [hunk 8]
```diff
@@ -55,18 +56,31 @@ impl TryFrom<Vec<u8>> for InterpreterCallerMeta {
     }
 }
 
+impl TryFrom<&[u8]> for InterpreterCallerMeta {
+    type Error = anyhow::Error;
```
Comment (thedavidmeister) [inline cli/src/meta/types/interpreter_caller/v1.rs:60]: don't use anyhow for lib logic i think

File: cli/src/meta/types/interpreter_caller/v1.rs [hunk 9]
```diff
@@ -55,18 +56,31 @@ impl TryFrom<Vec<u8>> for InterpreterCallerMeta {
...
+    fn try_from(value: &[u8]) -> Result<Self, Self::Error> {
+        match serde_json::from_slice::<Self>(value).map_err(anyhow::Error::from) {
+            Ok(t) => match t.validate().map_err(anyhow::Error::from) {
```
Comment (thedavidmeister) [inline cli/src/meta/types/interpreter_caller/v1.rs:63]: why is there a match _and_ a map err?

File: cli/src/meta/types/solidity_abi/v2.rs [hunk 10]
```diff
@@ -49,6 +52,19 @@ impl TryFrom<Vec<u8>> for SolidityAbiMeta {
     }
 }
 
+impl TryFrom<&[u8]> for SolidityAbiMeta {
+    type Error = anyhow::Error;
+    fn try_from(value: &[u8]) -> Result<Self, Self::Error> {
+        match serde_json::from_slice::<Self>(value).map_err(anyhow::Error::from) {
+            Ok(t) => match t.validate().map_err(anyhow::Error::from) {
```
Comment (thedavidmeister) [inline cli/src/meta/types/solidity_abi/v2.rs:59]: same as above

File: subgraph/flake.nix (hunk 5 above)
Comment (rouzwelt) [reply subgraph/flake.nix:28]: This is something @NanezX should look into probably

File: cli/src/meta/types/interpreter_caller/v1.rs (hunk 8 above)
Comment (rouzwelt) [reply cli/src/meta/types/interpreter_caller/v1.rs:60]: using anyhow for lib implementation was already implemented for structs/types of this repo before this PR, I just tried to keep it the same for newly added types/structs implementations

File: subgraph/flake.nix (hunk 5 above)
Comment (NanezX) [reply subgraph/flake.nix:28]: Where did you get that? Currently we don't use npx or mustache for generating the YAML with the address. We use rain.cli for that

Comment (rouzwelt) [reply subgraph/flake.nix:28]: @NanezX can you post the link to that please, thanks

**Pull request merged**